<?xml version="1.0" encoding="UTF-8"?>

<launch>

  <!-- This node will be used to convert the /people topic, which is 
  published by the open source leg tracker (people_velocity_tracker), 
  into the /tracked_persons topic, so that in the next it can be used 
  in the people and group tracking pipelines from the SPENCER-Project. 
  These two topics have different message structures. If you want to 
  use a different people tracker, you can modify this node to convert 
  your topic into the /tracked_persons topic so that it can be used in 
  the same pipelines from the SPENCER-Project. -->
  <node pkg="mybot_follower" name="people_to_tracked_person" type="people_to_tracked_persons.py"/>
  
  <!-- Start the nodes from SPENCER-Project to get spatial relations between people and start group tracking-->
  <include file="$(find mybot_follower)/launch/mybot_group_tracking.launch"/>

  <!-- The person closest to the robot in each group will be kept. -->
  <node pkg="mybot_follower" name="tracked_persons_filter" type="tracked_persons_filter.py"/>

  <!-- Find the most suitable group and broadcast the position of the person closest to the robot in this group. 
  Set the location of the destination, the maximum moving speed of the robot, and the maximum angle difference here.-->
  <node pkg="mybot_follower" name="mybot_people_tf_broadcaster" type="mybot_people_tf_broadcaster.py">
    <param name="goal_x" value="40.0"/> 
    <param name="goal_y" value="0.0"/>
    <param name="max_robot_velocity" value="2.0"/>
    <param name="max_direction_difference" value="0.5236"/> <!-- The angle between the moving direction of the human 
							    and the direction of the line connecting the robot to the 
							    goal must not be greater than this value. -->  
  </node>

  <!-- Move the robot to this position. -->
  <node pkg="mybot_follower" name="mybot_follower" type="mybot_follower.py"/>

</launch>
